pkg/operator/cluster/controller.go

```diff
@@ -180,7 +182,62 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 }
 
 func (c *ClusterController) onDelete(obj interface{}) {
-	logger.Infof("deleting a cluster not implemented")
+	clusterOrig := obj.(*Cluster)
+
+	// NEVER modify objects from the store. It's a read-only, local cache.
+	// Use scheme.Copy() to make a deep copy of original object.
+	copyObj, err := c.scheme.Copy(clusterOrig)
+	if err != nil {
+		logger.Errorf("failed to create a deep copy of cluster object: %v\n", err)
+		return
+	}
+	cluster := copyObj.(*Cluster)
+
+	logger.Infof("cluster in namespace %s is being deleted.  Waiting for agents to perform cleanup operations.", cluster.Namespace)
+
+	operatorNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+	retryCount := 0
```

Comment:
should you have these in a const?

Reply:
i swear there was a reusable Retry function somewhere in the operator package that does the retry looping around the function that you provide it, but i could not find it for the life of me.   Does it still exist?

Reply:
@travisn confirmed that there used to be a retry function but it was refactored out as no longer needed with the big CRD refactor.  I have made these consts, and also injectable into this function so that a very small sleep time can be used for unit tests.